src/main/java/com/microsoft/graph/httpcore/ChaosHttpHandler.java

```diff
@@ -0,0 +1,52 @@
+package com.microsoft.graph.httpcore;
+
+import java.io.IOException;
+import java.util.concurrent.ThreadLocalRandom;
+
+import com.microsoft.graph.httpcore.middlewareoption.MiddlewareType;
+import com.microsoft.graph.httpcore.middlewareoption.TelemetryOptions;
+
+import okhttp3.Interceptor;
+import okhttp3.MediaType;
+import okhttp3.Protocol;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+public class ChaosHttpHandler implements Interceptor {
+	
+	public final MiddlewareType MIDDLEWARE_TYPE = MiddlewareType.RETRY;
```

Comment:
nit: mixed tabs and spaces.

Most of the code I have seen in msgraph repos are using spaces.

I don't know your preference, but even if you prefer tabs, please make it all tabs within a specific file, because different tools assign different widths to tabs. For example, Github shows little wider than 4 spaces which makes it harder to follow when we have mixed tabs and spaces.

Reply:
I usually go with tabs. I'm not sure why but my tabs seem to be replaced by spaces on properties/fields.... Maybe because I copied the file from another handler to begin with. Do you know any command that normalizes tabbing in vs code?

Reply:
VS had this but I don't know if they have it for VSCode: https://marketplace.visualstudio.com/items?itemName=VisualStudioPlatformTeam.FixMixedTabs

I think these are the relevant settings for VSCode: https://stackoverflow.com/a/29972553

And on a funny note: https://stackoverflow.blog/2017/06/15/developers-use-spaces-make-money-use-tabs/ ;)